lighthouse-cli/test/smokehouse/smokehouse-report.js

```diff
@@ -102,45 +102,74 @@ function findDifference(path, actual, expected) {
   return null;
 }
 
+/**
+ * @param {string} name
+ * @param {any} actualResult
+ * @param {any} expectedResult
+ * @returns {Smokehouse.Assertion}
+ */
+function makeAssertion(name, actualResult, expectedResult) {
+  const diff = findDifference(name, actualResult, expectedResult);
+
+  return {
+    category: name,
+    actual: actualResult,
+    expected: expectedResult,
+    equal: !diff,
+    diff,
+  };
+}
+
 /**
  * Collate results into comparisons of actual and expected scores on each audit.
- * @param {Smokehouse.ExpectedLHR} actual
- * @param {Smokehouse.ExpectedLHR} expected
- * @return {Smokehouse.LHRComparison}
+ * @param {Smokehouse.ExpectedRunResult} actual
+ * @param {Smokehouse.ExpectedRunResult} expected
+ * @return {Smokehouse.RunComparison}
  */
 function collateResults(actual, expected) {
-  const auditNames = Object.keys(expected.audits);
-  const collatedAudits = auditNames.map(auditName => {
-    const actualResult = actual.audits[auditName];
-    if (!actualResult) {
-      throw new Error(`Config did not trigger run of expected audit ${auditName}`);
-    }
+  /** @type {Smokehouse.Assertion[]} */
+  const artifactAssertions = [];
+  if (expected.artifacts) {
+    Object.keys(expected.artifacts).map(artifactName => {
+      const actualResult = /** @type {any} */ (actual.artifacts)[artifactName];
+      if (!actualResult) {
+        throw new Error(`Config run did not generate artifact ${artifactName}`);
+      }
 
-    const expectedResult = expected.audits[auditName];
-    const diff = findDifference(auditName, actualResult, expectedResult);
+      const expectedResult = /** @type {any} */(expected.artifacts)[artifactName];
+      return makeAssertion(artifactName + ' artifact', actualResult, expectedResult);
+    });
+  }
 
-    return {
-      category: auditName,
-      actual: actualResult,
-      expected: expectedResult,
-      equal: !diff,
-      diff,
-    };
-  });
+  /** @type {Smokehouse.Assertion[]} */
+  const auditAssertions = [];
+  if (expected.lhr.audits) {
+    Object.keys(expected.lhr.audits).map(auditName => {
+      const actualResult = actual.lhr.audits[auditName];
+      if (!actualResult) {
+        throw new Error(`Config did not trigger run of expected audit ${auditName}`);
+      }
+
+      const expectedResult = expected.lhr.audits[auditName];
+      return makeAssertion(auditName + ' audit', actualResult, expectedResult);
+    });
+  }
+
+  const assertions = [...artifactAssertions, ...auditAssertions];
 
   return {
-    audits: collatedAudits,
+    assertions,
     errorCode: {
       category: 'error code',
-      actual: actual.errorCode,
-      expected: expected.errorCode,
-      equal: actual.errorCode === expected.errorCode,
+      actual: actual.lhr.errorCode,
```

Comment:
now that we are fixing things, this isn't really in the LHR, so makes sense to move out at top level